src/main/java/com/google/sps/Database.java

```diff
@@ -0,0 +1,56 @@
+package com.google.sps;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.FirestoreOptions;
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.WriteResult;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.SetOptions;
+
+import java.util.*;
+import java.io.*; 
+
+public class Database {
+    private Firestore db;
+
+    public void initialize() throws Exception {
+        GoogleCredentials credentials = GoogleCredentials.getApplicationDefault();
+        // Initialize instance of Firestore
+        FirestoreOptions firestoreOptions = FirestoreOptions.getDefaultInstance().toBuilder()
+                .setProjectId("spring21-sps-33").setCredentials(credentials).build();
+        db = firestoreOptions.getService();
+    }
+
+    public void addData(String filePath) throws Exception {
+        File fecData = new File(filePath);
+        
+        Scanner input = new Scanner(fecData); 
+
+        while (input.hasNextLine()){
+            String can = input.nextLine(); 
+            String[] stats = can.split("\\|"); 
+
+            CandidateStats canStats = new CandidateStats(stats[0], stats[18], stats[19], stats[4], stats[11], 
```

Comment:
Your code looks great and I'm glad we got it working yesterday.
The only improvement I can suggest is including a comment above line 34 of Database.java that serves as a key/legend for CandidateStats so that it's clear which index of stats represents what info for candidates.

Reply:
The comment is a good suggestion. A syntax I'd like you to use is new CandidateStats(/*name= */stats[0] .... for each of the parameters.

Reply:
Oh, this is helpful! I didn't know you could do that. I'll make those changes now!